### PR TITLE
feat(CI): separated build pch and no-pch

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Add support pch for all `Ubuntu` build
- Make separated no-pch build with (Ubuntu latest and clang)
- Reduced compile time for all build (Less workload for GitHub actions)

### CODE REVIEW

File: apps/ci/ci-conf-no-pch.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
```
Comment (FrancescoBorzi) [inline apps/ci/ci-conf-no-pch.sh:1]: do we really need to duplicate this??

Comment (Winfidonarleyan) [reply apps/ci/ci-conf-no-pch.sh:1]: I don't know how to do otherwise :/

**Comment by locus313** [comment]
Enabling pch is going to make the ci builds take much longer.  We should not enable it for the whole core-build or docker build. Between @Yehonal  and me with purposely disable pch to speed up the build times. If we do enable a pch build it should be in a seprate workflow that does not run automatically. (like the macos, windows, docker workflows)

**Comment by Winfidonarleyan** [comment]
How can it take longer to compile a core with pch?

`succeeded 6 hours ago in 53m 40s`
https://github.com/azerothcore/azerothcore-wotlk/pull/6008/checks?check_run_id=2649285455

**Comment by locus313** [comment]
Take a look at the info in this pr:

https://github.com/azerothcore/azerothcore-wotlk/pull/5427

Build times where greatly  reduced when disabling pch.

When I did some research on the issue it seems to be mainly due to how ccache handles the matching.

File: .github/workflows/core_build_no_pch.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.compiler }}
+    env:
+      COMPILER: ${{ matrix.compiler }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: /home/runner/.ccache
```
Comment (locus313) [inline .github/workflows/core_build_no_pch.yml:25]:
```suggestion
          path: var/ccache
```

Comment (locus313) [reply .github/workflows/core_build_no_pch.yml:25]: @Yehonal built in support for ccache directly into the scripts so to properly save the cache you need to change the path to this.

**Comment by Winfidonarleyan** [comment]
No need for this repo

**Pull request closed without merging**